### PR TITLE
Change due date format for user management details

### PR DESCRIPTION
**Pull request by Shenoy07**
# Overview

## **PR for issue #472 .**

The aim is to display the number of days left before the due date of a task. On hovering we have to display the actual due date as well.
Currently, this is how the tasks>duedate section looks like.
 
image

# Goals:
- The Tasks > DueDate section should show the number of days remaining for the task to end.
- When we hover we want the actual due date to pop up as well.

# Steps taken:

## 1. Got Access as a super user:
- Manipulate the if condition to get access as a super user. 
 
image

## 2. Created a function to calculate the number of days remaining before the due date:

- The below flow was followed in the function.
image

## 3. Made the necessary CSS changes for the hover element.
image

## 4. Created test files:
- Understood how to test consecutive selectors using image https://pptr.dev/
- Created mock data for the one of API calls as the other one was made as a part of #476
  - https://api.realdevsquad.com/tasks/username (to get the user task details) [created in the current PR #501]
  - https://api.realdevsquad.com/users/self (to pass as a super user and get to see content) [created inPR #476]

## 5. Checked Test files:
 - Here is the video of the test.

https://github.com/Real-Dev-Squad/website-dashboard/assets/31709147/32704b80-de51-4758-aeaf-6d25ca92fe9d

 - All test cases passing.

image

## 6. Post-making changes to the due date section we see the following change.

image

### CODE REVIEW

**Comment by gauravsinhaweb** [comment]
Is this feature behind the feature flag?

File: users/details/script.js [hunk 1]
```diff
@@ -350,6 +371,28 @@ function generateReadableDateFromSecondsTimeStamp(timeStamp) {
...
+
+function generateDaysToGo(dateStr, status) {
+  const inputDate = new Date(dateStr);
+
+  const now = new Date();
+  const offset = 330 * 60 * 1000;
+  const currentDate = new Date(now.getTime() + offset);
+  const diff = inputDate - currentDate; // Calculates the difference in milliseconds
+  if (diff <= 0 && status == 'COMPLETED') {
+    return 'Task Completed Within Deadline';
```
Comment (iamitprakash) [inline users/details/script.js:384]: what about below message. 

"Task has been completed within Committed timeline."

File: users/details/script.js [hunk 2]
```diff
@@ -350,6 +371,28 @@ function generateReadableDateFromSecondsTimeStamp(timeStamp) {
...
+  } else if (diff <= 0) {
+    return 'Deadline Passed'; // Due date is in the past
+  } else if (diff > 0 && diff < 24 * 60 * 60 * 1000) {
+    return 'Less Than a Day Remaining'; // Less than a day remaining
```
Comment (iamitprakash) [inline users/details/script.js:388]: Less than a Day is remaining.

with Yellow font color

File: users/details/script.js [hunk 3]
```diff
@@ -350,6 +371,28 @@ function generateReadableDateFromSecondsTimeStamp(timeStamp) {
...
+  } else {
+    const daysRemaining = Math.floor(diff / (24 * 60 * 60 * 1000)); // Calculate the days remaining
+    return daysRemaining === 1
+      ? '1 day remaining'
+      : daysRemaining + ' days remaining'; // Handle singular and plural for 1 day and more than 1 day
```
Comment (iamitprakash) [inline users/details/script.js:393]: Day 'D' upper case please

File: users/details/script.js [hunk 4]
```diff
@@ -350,6 +371,28 @@ function generateReadableDateFromSecondsTimeStamp(timeStamp) {
...
+console.log(generateReadableDateFromSecondsTimeStamp('1688445662'));
+
+function generateDaysToGo(dateStr, status) {
+  const inputDate = new Date(dateStr);
+
+  const now = new Date();
+  const offset = 330 * 60 * 1000;
+  const currentDate = new Date(now.getTime() + offset);
+  const diff = inputDate - currentDate; // Calculates the difference in milliseconds
+  if (diff <= 0 && status == 'COMPLETED') {
```
Comment (iamitprakash) [inline users/details/script.js:383]: can we use enum to compare status value

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: users/details/script.js (hunk 2 above)
Comment (Shenoy07) [reply users/details/script.js:388]: image

For some reason, I am finding **Yellow** to be a bit difficult to read, I tried **orange** as well, I think it provides better readability, but I know you suggested **yellow** as it is supposed to be a **warning sign**.
image

Please let me know which one should I go ahead with.

File: users/details/script.js (hunk 1 above)
Comment (Shenoy07) [reply users/details/script.js]: successfully made the changes.
image

File: users/details/script.js (hunk 3 above)
Comment (Shenoy07) [reply users/details/script.js]: Implemented
image

Comment (Shenoy07) [reply users/details/script.js]: missed one spot, updating it as well

File: users/details/script.js (hunk 2 above)
Comment (Shenoy07) [reply users/details/script.js:392]: Currently I have put yellow.

File: users/details/script.js (hunk 3 above)
Comment (Shenoy07) [reply users/details/script.js]: Updated.
image

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Review by iamitprakash — APPROVED** [review]

**Review by iamitprakash — CHANGES_REQUESTED** [review]
Please add feature flag

**Comment by iamitprakash** [comment]
@gauravsinhaweb @prakashchoudhary07  can you  please @Shenoy07  to understand feature flag into PR

File: users/details/script.js [hunk 5]
```diff
@@ -350,6 +375,28 @@ function generateReadableDateFromSecondsTimeStamp(timeStamp) {
   //created function for readable date format
   return new Date(timeStamp * 1000).toDateString(); // new function because we are getting the value in seconds and not milliseconds
 }
+console.log(generateReadableDateFromSecondsTimeStamp('1688445662'));
```
Comment (iamitprakash) [inline users/details/script.js:378]: Why this Console.log

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: users/details/script.js (hunk 5 above)
Comment (Achintya-Chatterjee) [inline users/details/script.js:378]: Please explain why you put console log here?

**Comment by gauravsinhaweb** [comment]
> @gauravsinhaweb @prakashchoudhary07  can you  please @Shenoy07  to understand feature flag into PR

We (@Shenoy07 ) had a discussion about FF over voice chat, shown a demo of it and asked him to make necessary changes.

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]
remove console log from user details

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Comment by Shenoy07** [comment]
Added feature flag:

Before:
image

After:
image

File: users/details/script.js (hunk 5 above)
Comment (Shenoy07) [reply users/details/script.js]: Sorry about this. Changed.

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]
can you please sort out the undefined

**Comment by Shenoy07** [comment]
Before dev=false:
image

After dev=true:
image

**Review by iamitprakash — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Pull request merged**